iOS/README.rst

```diff
@@ -0,0 +1,294 @@
+the API used on these devices is identical, the ABI is different - you need to
+link against different libraries for an iOS device build (``iphoneos``) or an
+iOS simulator build (``iphonesimulator``).
+
+Apple uses the ``XCframework`` format to allow specifying a single dependency
+that supports multiple ABIs. An ``XCframework`` is a wrapper around multiple
+ABI-specific frameworks that share a common API.
+
+iOS can also support different CPU architectures within each ABI. At present,
+there is only a single supported architecture on physical devices - ARM64.
+However, the *simulator* supports 2 architectures - ARM64 (for running on Apple
+Silicon machines), and x86_64 (for running on older Intel-based machines).
+
+To support multiple CPU architectures on a single platform, Apple uses a "fat
+binary" format - a single physical file that contains support for multiple
+architectures. It is possible to compile and use a "thin" single architecture
+version of a binary for testing purposes; however, the "thin" binary will not be
+portable to machines using other architectures.
+
+Building a single-architecture framework
+----------------------------------------
+
+The Python build system will create a ``Python.framework`` that supports a
+*single* ABI with a *single* architecture. Unlike macOS, iOS does not allow a
+framework to contain non-library content, so the iOS build will produce a
+``bin`` and ``lib`` folder in the same output folder as ``Python.framework``.
+The ``lib`` folder will be needed at runtime to support the Python library.
+
+If you want to use Python in a real iOS project, you need to produce multiple
+``Python.framework`` builds, one for each ABI and architecture. iOS builds of
+Python *must* be constructed as framework builds. To support this, you must
+provide the ``--enable-framework`` flag when configuring the build. The build
+also requires the use of cross-compilation. The minimal commands for building
+Python for the ARM64 iOS simulator will look something like::
+
+  $ export PATH="`pwd`/iOS/Resources/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin"
+  $ ./configure \
+        AR=arm64-apple-ios-simulator-ar \
+        CC=arm64-apple-ios-simulator-clang \
+        CPP=arm64-apple-ios-simulator-cpp \
+        CXX=arm64-apple-ios-simulator-clang \
+        --enable-framework=/path/to/install \
+        --host=arm64-apple-ios-simulator \
+        --build=arm64-apple-darwin \
+        --with-build-python=/path/to/python.exe
+  $ make
+  $ make install
+
+In this invocation:
+
+* ``iOS/Resources/bin`` has been added to the path, providing some shims for the
+  compilers and linkers needed by the build. Xcode requires the use of ``xcrun``
+  to invoke compiler tooling; however, ``xcrun`` embeds user- and
+  version-specific paths into the sysconfig data, which limits the portability
+  of the compiled Python. It also requires that compiler variables like ``CC``
```

Comment:
I'm not sure I follow this – there are no user- or version-specific paths in the wrapper scripts, so why would there be any in the `CC` variable if it was pointed at `xcrun` directly?

Reply:
So - it depends a little on how you end up using `xrcun`. 

If you set `CC=xcrun --sdk iphoneos clang --target arm64-apple-iphoneos`, you've put spaces in your CC definition.

However, if you pre-evaluate the `xcrun` invocation, and use `CC=$(xcrun --sdk iphoneos clang --target arm64-apple-iphoneos)`, you can end up with version numbers and/or user-specific paths in your CC definition (depending on exactly how your Xcode install is set up).

I've clarified this paragraph; hopefully it's more clear now.